### PR TITLE
[KeyVault] Updating internal to go1.18

### PR DESCRIPTION
**Pull request by seankane-msft**
This will need a release after the azcore release

### CODE REVIEW

**Review by chlowell — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]

File: sdk/keyvault/internal/challenge_policy.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
-//go:build go1.16
-// +build go1.16
+//go:build go1.18
+// +build go1.18
```
Comment (chlowell) [inline sdk/keyvault/internal/challenge_policy.go:2]: The `+build` tags are only needed for Go < 1.17, so they're redundant now.

Comment (jhendrixMSFT) [reply sdk/keyvault/internal/challenge_policy.go:2]: I'll update the code generator with this change shortly.

Comment (jhendrixMSFT) [reply sdk/keyvault/internal/challenge_policy.go:2]: I will point out though that removing them will break building with older versions of Go.

Comment (seankane-msft) [reply sdk/keyvault/internal/challenge_policy.go:2]: This uses the `to.Ptr` method, it won't build with older versions of Go anyways.

Comment (jhendrixMSFT) [reply sdk/keyvault/internal/challenge_policy.go:2]: To clarify, I'm talking about track 1.  We had this problem in the past, see https://github.com/Azure/azure-sdk-for-go/issues/6846.  We might need to keep these until track 1 is out of `main` (needs investigation).

Comment (seankane-msft) [reply sdk/keyvault/internal/challenge_policy.go:2]: I'll keep these for now

**Pull request merged**